Gems/ROS2/Code/Source/RobotImporter/RobotImporterWidget.cpp

```diff
@@ -78,12 +80,35 @@ namespace ROS2
 
     void RobotImporterWidget::OpenUrdf()
     {
-        m_urdfPath = AZStd::string(m_fileSelectPage->getFileName().toUtf8().constData());
+        QString report;
         if (!m_urdfPath.empty())
         {
-            AZ_Printf("Wizard", "Loading URDF file : %s", m_urdfPath.c_str());
-            m_parsedUrdf = UrdfParser::ParseFromFile(m_urdfPath);
-            QString report;
+            if (m_urdfPath.ends_with("xacro"))
+            {
+                Utils::xacro::ExecutionOutcome outcome = Utils::xacro::ParseXacro(m_urdfPath, m_params);
+                if (outcome)
+                {
+                    m_parsedUrdf = outcome.m_urdfHandle;
+                    report += "# " + tr("XACRO execution succeed") + "\n";
+                }
+                else
+                {
+                    report += "# " + tr("XACRO parsing failed") + "\n";
+                    report += "\n\n" + tr("Command called : \n'") + QString::fromUtf8(outcome.m_called.data()) + "'";
+                    report += "\n\n" + tr("Process failed with code : ") + QString::number(outcome.m_returnCode);
+                    report += "\n\n" + tr("error output") + " :\n\n";
+                    report += QString::fromLocal8Bit(outcome.m_logErrorOutput.data(), static_cast<int>(outcome.m_logErrorOutput.size()));
```

Comment:
nit: Should add an additional '\n'
```suggestion
                    report += QString::fromLocal8Bit(outcome.m_logErrorOutput.data(), static_cast<int>(outcome.m_logErrorOutput.size())) + "\n";
```